tensorflow_lite_support/ios/task/vision/sources/TFLImageSearcher.h

```diff
@@ -0,0 +1,113 @@
+
+/**
+ * Options controlling the behavior of the embedding model specified in the
+ * base options.
+ */
+@property(nonatomic, copy) TFLEmbeddingOptions *embeddingOptions;
+
+/**
+ * Options specifying the index to search into and controlling the search behavior.
+ */
+@property(nonatomic, copy) TFLSearchOptions *searchOptions;
+
+/**
+ * Initializes a new `TFLImageSearcherOptions` with the absolute path to the model file
+ * stored locally on the device, set to the given the model path.
+ *
+ * @discussion The external model file, must be a single standalone TFLite file. It could be packed
+ * with TFLite Model Metadata[1] and associated files if exist. Fail to provide the necessary
+ * metadata and associated files might result in errors. Check the [documentation]
+ * (https://www.tensorflow.org/lite/convert/metadata) for each task about the specific requirement.
+ *
+ * @param modelPath An absolute path to a TensorFlow Lite model file stored locally on the device.
+ *
+ * @return An instance of `TFLImageSearcherOptions` initialized to the given
+ * model path.
+ */
+- (instancetype)initWithModelPath:(NSString *)modelPath;
+
+@end
+
+/**
+ * A TensorFlow Lite Task Image Searcher.
+ */
+NS_SWIFT_NAME(ImageSearcher)
+@interface TFLImageSearcher : NSObject
+
+/**
+ * Creates a new instance of `TFLImageSearcher` from the given `TFLImageSearcherOptions`.
+ *
+ * @param options The options to use for configuring the `TFLImageSearcher`.
+ * @param error An optional error parameter populated when there is an error in initializing
+ * the image searcher.
+ *
+ * @return A new instance of `ImageSearcher` with the given options. `nil` if there is an error
+ * in initializing the image searcher.
+ */
++ (nullable instancetype)imageSearcherWithOptions:(TFLImageSearcherOptions *)options
+                                            error:(NSError **)error
+    NS_SWIFT_NAME(searcher(options:));
+
++ (instancetype)new NS_UNAVAILABLE;
+
+/**
+ * Performs embedding extraction on the given GMLImage, followed by nearest-neighbor search in the
+ * index.
+ *
+ * @discussion This method currently supports searching on only the following types of images:
+ * 1. RGB and RGBA images for `GMLImageSourceTypeImage`.
+ * 2. kCVPixelFormatType_32BGRA for `GMLImageSourceTypePixelBuffer` and
+ *    `GMLImageSourceTypeSampleBuffer`. If you are using `AVCaptureSession` to setup
+ *    camera and get the frames for inference, you must request for this format
+ *    from AVCaptureVideoDataOutput. Otherwise your inference results will be wrong.
+ *
+ * @param image An image on which embedding extraction is to be performed, followed by
+ * nearest-neighbor search in the index, represented as a `GMLImage`.
+ *
+ * @return A `TFLSearchResult`. `nil` if there is an error encountered during embedding extraction
+ * and nearest neighbor search. Please see `TFLSearchResult` for more details.
+ */
+- (nullable TFLSearchResult *)searchInGMLImage:(GMLImage *)image
```

Comment:
It should be `searchWithGMLImage`. "with" rather than "in" because we don't search in the image but find similar images in a pre-indexed database.